sarif/main.ts

```diff
@@ -0,0 +1,174 @@
+import { PhylumApi } from 'phylum';
+import { parse } from "https://deno.land/std@0.156.0/flags/mod.ts";
+
+type Dependency = {
+    purl: string;
+    registry: string;
+    name: string;
+    version: string;
+    issues: Issue[];
+};
+
+type Issue = {
+    title: string;
+    tag: string;
+    domain: string;
+    severity: string;
+    description: string;
+};
+
+type InputJson = {
+    dependencies: Dependency[];
+};
+
+type Project = {
+    name: string;
+    id: string;
+    updated_at: string;
+    created_at: string;
+    ecosystems: string[];
+    group_name: string;
+};
+
+const accessToken = await PhylumApi.getAccessToken();
+
+/**
+ * Given a Phylum JSON response, convert it into a valid SARIF JSON.
+ */
+const convertToSarif = (input: InputJson) => {
+    const sarif: any = {
+        "$schema": "https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/schemas/sarif-schema-2.1.0.json",
+        "version": "2.1.0",
+        "runs": [
+            {
+                "tool": {
+                    "driver": {
+                        "name": "Phylum",
+                        "version": "1.0.0",
+                        "informationUri": "https://phylum.io",
+                        "rules": [] // This will be populated later
+                    }
+                },
+                "results": [] // This will be populated later
+            }
+        ]
+    };
```

Comment:
Can we reverse this so we build the rules and results first and then build the sarif object containing it? It seems like it'd be cleaner not having to modify `sarif.runs[0]` all the time.

Reply:
done!